frontend/src/Editor/Inspector/Components/Table.jsx

```diff
@@ -730,7 +730,7 @@ class Table extends React.Component {
 
     const conditionalOptions = [
       { name: 'clientSidePagination', condition: !serverSidePagination },
-      { name: 'displaySearchBox', condition: displaySearchBox },
+      !serverSidePagination && { name: 'displaySearchBox', condition: true },
```

Comment:
Let's keep same type of object in the array, this will add `false` as second element of the array, if server side pagination is enabled.
`!serverSidePagination && conditionalOptions.push({ name: 'displaySearchBox', condition: true })` Conditionally push the value.